xwiki-platform-core/xwiki-platform-menu/xwiki-platform-menu-ui/src/main/resources/Menu/MenuMacro.xml

```diff
@@ -809,10 +815,12 @@ require(['jquery','xwiki-l10n!menu-ui-translation-keys'], function($, l10n) {
         li {
           /* Text inside menu */
           color: @dropdown-link-color;
+          display: flex;
+          align-items: center;
+          padding: 3px 20px;
           /* Links inside menu */
           a {
             display: block;
-            padding: 3px 20px;
```

Comment:
Why is moving the padding necessary? Doesn't this make the links smaller and thus harder to click?

Reply:
This ticket was raised because I didn't take into account the fact that users could put multiple elements in the top level menu, I want to make sure there is no issues if submenus are used the same way. If we assume that there is just one element per submenus (anchor or a text title for the subsection), this change is not needed.

Because of the padding on the links, we can't use multiple elements in a submenu without messing up the presentation.
Double margins, no margins, etc...
![20922-BothPaddings](https://github.com/xwiki/xwiki-platform/assets/28761965/a4775e78-df35-4445-822e-dd5b149035ee)

Icons can be used **if they are contained by the anchor** but that's it (Note: I'm pretty sure this ticket would have not been raised if the icons were part of the anchors in the top menu).
It makes the style much more difficult because we have to take into account a lot of cases: anchor by itself, no anchor, anchor with icon before, anchor wiht icon after, all of those with and without a flyout submenu (with the carret to keyboard control it).

Is it worth the resource to add event listeners on all the submenu items that redirect the _click_ event to the first link in their content? Should we just assume that there is only one item in a submenu line, and be done with these style complexities.

Reply:
Extra notes: The events for the menu/submenu flying out are on the hover of the container element. So the links being small would not impact all steps of navigation but only the last one (aka. click the anchor to reach the page, and not all the steps to get this anchor visible and/or in focus).

The :hover background is only set for the link, so it's not looking the best in some interactions.

Reply:
Created [a separate ticket](https://jira.xwiki.org/browse/XWIKI-20930) to extend on this reflection and avoid blocking the resolution of the actual issue at stakes here.